File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/ResponsesApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class ResponsesApi {
    private final HttpClient client;
    
    public ResponsesApi(HttpClient client) {
        this.client = client;
    }

    /** Cancel response */
    public CancelResponseResponse cancelResponse(String response_id) throws Exception {
        return (CancelResponseResponse) client.post(ApiPaths.aiPath("/responses/" + response_id + "/cancel"), null);
    }

    /** Create response */
    public CreateResponsePostResponse createResponse(CreateResponsePostRequest body) throws Exception {
        return (CreateResponsePostResponse) client.post(ApiPaths.aiPath("/responses"), body);
    }

    /** List response input items */
    public ListResponseInputItemsGetResponse listResponseInputItems(String response_id) throws Exception {
        return (ListResponseInputItemsGetResponse) client.get(ApiPaths.aiPath("/responses/" + response_id + "/input_items"));
    }

    /** Get response */
    public GetResponseGetResponse getResponse(String response_id) throws Exception {
        return (GetResponseGetResponse) client.get(ApiPaths.aiPath("/responses/" + response_id + ""));
    }

    /** Delete response */
    public DeleteResponseDeleteResponse deleteResponse(String response_id) throws Exception {
        return (DeleteResponseDeleteResponse) client.delete(ApiPaths.aiPath("/responses/" + response_id + ""));
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/ResponsesApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class ResponsesApi(private val client: HttpClient) {

    /** Cancel response */
    suspend fun cancelResponse(response_id: String): CancelResponseResponse? {
        return client.post(ApiPaths.aiPath("/responses/$response_id/cancel"), null) as? CancelResponseResponse
    }

    /** Create response */
    suspend fun createResponse(body: CreateResponsePostRequest): CreateResponsePostResponse? {
        return client.post(ApiPaths.aiPath("/responses"), body) as? CreateResponsePostResponse
    }

    /** List response input items */
    suspend fun listResponseInputItems(response_id: String): ListResponseInputItemsGetResponse? {
        return client.get(ApiPaths.aiPath("/responses/$response_id/input_items")) as? ListResponseInputItemsGetResponse
    }

    /** Get response */
    suspend fun getResponse(response_id: String): GetResponseGetResponse? {
        return client.get(ApiPaths.aiPath("/responses/$response_id")) as? GetResponseGetResponse
    }

    /** Delete response */
    suspend fun deleteResponse(response_id: String): DeleteResponseDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/responses/$response_id")) as? DeleteResponseDeleteResponse
    }
}
